src/lib.rs

```diff
@@ -40,24 +55,47 @@ impl TsnSocket {
     }
 }
 
-const CONTROL_SOCK_PATH: &str = "/var/run/tsn.sock";
-
-fn send_cmd(command: String) -> Result<String, std::io::Error> {
-    let mut stream = UnixStream::connect(CONTROL_SOCK_PATH)?;
-    stream.write_all(command.as_bytes())?;
-    let mut msg = String::new();
-    stream.read_to_string(&mut msg)?;
-    Ok(msg)
-}
-
-fn create_vlan(ifname: &str, vlanid: u16) -> Result<String, std::io::Error> {
-    let command = format!("create {} {}\n", ifname, vlanid);
-    send_cmd(command)
+fn create_vlan(ifname: &str, vlanid: u16) -> Result<i32, i32> {
```

Comment:
It could be better if `Result<(), String>` (Also on vlan::{create_vlan, delete_vlan}), But passing this time.